src/java/org/apache/cassandra/db/streaming/CassandraStreamManager.java

```diff
@@ -126,6 +127,8 @@ else if (pendingRepair == ActiveRepairService.NO_PENDING_REPAIR)
                 return sstables;
             }).refs);
 
+            // Persistent memtables will not flush, make an sstable with their data.
+            cfs.writeAndAddMemtableRanges(session.getPendingRepair(), () -> Range.normalize(keyRanges), refs);
```

Comment:
Q: Does the flush that corresponds to this occur in `addTransferRanges()`? If so, it might be helpful to indicate that in a comment. (It's a little easier to make the connection in the validation iterator than here.)